test_scripts/Defects/8_2/3906/002_SDL_does_not_apply_PTU_via_PutFile_during_PTU_sequence.lua

```diff
@@ -0,0 +1,72 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/SmartDeviceLink/sdl_core/issues/3906
+---------------------------------------------------------------------------------------------------
+-- Description: SDL does not apply PTU provided via PutFile from mobile app during PTU sequence
+--
+-- Preconditions:
+-- 1. SDL and HMI are started
+-- 2. Mobile app is registered and activated
+-- SDL does:
+--  - start PTU and sends SDL.OnStatusUpdate(UPDATE_NEEDED, UPDATING) to HMI
+-- 3. SDL sends OnSystemRequest(PROPRIETARY) request to App
+-- Steps:
+-- 1. App sends PutFile request with updated policy table includes "HapticGroup" functional group
+-- SDL does:
+-- - not send SDL.OnStatusUpdate(UP_TO_DATE) notification to HMI
+-- - not send SDL.OnPermissionsChange notification to HMI
+-- - not send SDL.OnAppPermissionChanged notification to HMI
+-- - sends BasicCommunication.OnPutFile notification to HMI
+-- - respond PutFile(success = true, resultCode = "SUCCESS") to app
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require("test_scripts/Defects/8_2/3906/common_3906")
+
+--[[ Test Configuration ]]
+common.testSettings.restrictions.sdlBuildOptions = {{ extendedPolicy = { "PROPRIETARY", "EXTERNAL_PROPRIETARY" }}}
+
+--[[ Local Functions ]]
+local function policyTableUpdateProprietary(pPTUpdateFunc)
+  local ptuFileName = os.tmpname()
+  local requestId = common.getHMIConnection():SendRequest("SDL.GetPolicyConfigurationData",
+    { policyType = "module_config", property = "endpoints" })
+  common.getHMIConnection():ExpectResponse(requestId)
+  :Do(function()
+      common.getHMIConnection():SendNotification("BasicCommunication.OnSystemRequest",
+        { requestType = "PROPRIETARY", fileName = common.getPTSFilePath() })
+      local ptuTable = common.getPTUFromPTS()
+      ptuTable.policy_table.app_policies[common.getPolicyAppId()] = common.getAppDataForPTU()
+      if pPTUpdateFunc then
+        pPTUpdateFunc(ptuTable)
+      end
+      common.tableToJsonFile(ptuTable, ptuFileName)
+      local event = common.createEvent()
+      common.getHMIConnection():ExpectEvent(event, "PTU event")
+      common.getMobileSession():ExpectNotification("OnSystemRequest", { requestType = "PROPRIETARY" })
+      :Do(function()
+          common.getHMIConnection():ExpectRequest("VehicleInfo.GetVehicleData", { odometer = true }):Times(0)
+          common.getHMIConnection():ExpectNotification("SDL.OnStatusUpdate", { status = "UP_TO_DATE" }):Times(0)
```

Comment:
Maybe the good solution to call in this `Do` block the `putFile` function with required permissions